matching-notification-rest/src/main/java/org/phenotips/matchingnotification/rest/internal/DefaultMatchingNotificationResource.java

```diff
@@ -455,4 +456,38 @@ private boolean isNonexistingOrSelfMatch(PatientMatch match)
 
         return false;
     }
+
+    @Override
+    public Response getMatches(String referencePatientId, String referenceServerId, String matchedPatientId,
+        String matchedServerId)
+    {
+        if (StringUtils.isBlank(referencePatientId) || StringUtils.isBlank(matchedPatientId)) {
+            this.logger.error("One of the requested patient ids parameter is blank and thus not a valid input JSON");
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+
+        List<PatientMatch> matches = this.matchStorageManager.loadMatchesBetweenPatients(
+            referencePatientId, referenceServerId, matchedPatientId, matchedServerId);
+
+        JSONObject matchIDJson = new JSONObject();
+
+        if (!matches.isEmpty()) {
+            matchIDJson.put(MATCHID_STRING, getLastOutgoingMatchId(matches));
```

Comment:
I don't remember by now, but what if there are matches, but none of them are outgoing, should we return `null` or the id of the last incoming match instead? I forgot, where do we use this endpoint?

P.S. also I don't think we should be returning `null`-s, but thats what we do in `getLastOutgoingMatchId()` if there are no outgoing matches